Update Sentry <> Expo (#36231)
expo/expo: docs/pages/guides/using-sentry.mdx

<PlatformsSection title="Platform compatibility" android emulator ios simulator web />

> ## What you'll learn

Reviewer: This section should go above PlatformSection.

This guide covers three main aspects of integrating Sentry with your Expo projects:

> 1. [Installing and configuring Sentry](#install-and-configure-sentry) in your React Native application
> 2. Using Sentry with Expo services:
>    - [EAS Build](#usage-with-eas-build) for building your application
>    - [EAS Update](#usage-with-eas-update) for over-the-air updates
> 3. [Setting up the Sentry-Expo integration](#sentry-integration-with-eas) to view crash reports and session replays directly in your EAS dashboard

Reviewer: ```suggestion
1. [Install and configure Sentry](#install-and-configure-sentry) in your React Native app
2. Using Sentry with EAS:
   - [EAS Build](#usage-with-eas-build) for building your app
   - [EAS Update](#usage-with-eas-update) for over-the-air updates
3. [Setting up the Sentry-Expo integration](#sentry-integration-with-eas) to view crash reports and session replays directly in your Expo dashboard
```

- Let's update application instances to "app" for consistency
- We use Expo dashboard instead of EAS elsewhere in our docs

<Step label="2">

### Use the Sentry wizard to set up your project

> The easiest way to set up Sentry in your Expo React Native project is to use the Sentry wizard. This tool will automatically configure your project with the right settings.

Reviewer 2: ```suggestion
The easiest way to set up Sentry in your Expo project is to use the Sentry wizard. This tool will automatically configure your project with the right settings.
```

Run the following command in your project directory:

<Terminal cmd={['$ npx @sentry/wizard@latest -i reactNative']} />

The wizard will:

> 1. Install the required dependencies
> 2. Configure your project to use Sentry
> 3. Set up the Metro configuration automatically
> 4. Add the necessary initialization code to your app

Reviewer: ```suggestion
- Install the required dependencies
- Configure your project to use Sentry
- Set up the Metro configuration automatically
- Add the necessary initialization code to your app
```

> Follow the prompts in the wizard to complete the setup process. The wizard will ask for your **organization slug**, **project name**, and **auth token** that you noted earlier.

Reviewer: ```suggestion
Follow the prompts in the wizard to complete the setup process. The wizard will ask for your **organization slug**, **project name**, and **auth token** that you noted earlier.
```

If I've a Sentry account, I do not need to add the details like Organization Slug, project name, and auth token on my own and instead the wizard from the CLI just asks me to login to my Sentry account. I was wondering if this could be reflected here as well and also make Step 1 shorter, so that the user who has created a new account or already has an account does not need to provide these details.

Author: updated in my review to be more direct about what the wizard will do

> </Step>

Reviewer: A couple of lines below (line no 84), can you please update the step number to 3?

<Step label="2">

### Use the Sentry wizard to set up your project

The easiest way to set up Sentry in your Expo project is to use the Sentry wizard. This tool will automatically configure your project with the right settings.

Run the following command in your project directory:

<Terminal cmd={['$ npx @sentry/wizard@latest -i reactNative']} />

The wizard will:

- Install the required dependencies
- Configure your project to use Sentry
- Set up the Metro configuration automatically
- Add the necessary initialization code to your app

> Follow the prompts in the wizard to complete the setup process. The wizard will ask for your **organization slug**, **project name**, and **auth token** that you noted earlier.

Author: ```suggestion
Follow the prompts in the wizard to complete the setup process. The wizard will guide you to log in to your Sentry account and fetch all the correct information regarding your project.
```

Author: @amandeepmittal we can write it like this instead.

Reviewer: Yes, that sounds perfect.

</Collapsible>

> ## Sentry Integration with EAS

Reviewer: ```suggestion
## Sentry integration with Expo dashboard
```

- We use sentence case instead of title case as per our writing style guide: https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#headings
- Also, let's rename this section title. It is a bit confusing to find this wording at the end of the page even though the whole guide is meant integrate Sentry in Expo apps and EAS.
- I'd also recommend moving this section beofre "Usage with EAS Build" and "Usage with EAS Update"

> ## Sentry Integration with EAS

Reviewer: Also, when I am trying to connect my Sentry account to Expo account, I got stuck on the below screen. I was doing this to test the connection features as described below. Do you think I am missing any steps?

![CleanShot 2025-04-22 at 17 08 28@2x](https://github.com/user-attachments/assets/1b81721a-9740-4ec2-98ec-6a8a2a189031)

Reviewer: Thanks for sharing the context! I was lacking it.
The integration button is available to me in Expo dashboard.I have a staff account, which might be feature-flagged enabled on my end and that's why it might be showing under my Expo account. I'll re-check.

> The Sentry integration with Expo allows you to view crash reports and Session Replays for your Expo app deployments directly within your Expo Application Services (EAS) dashboard. This integration provides a direct link to Sentry stack traces with full context, session replays, and debugging capabilities.

Reviewer: ```suggestion
The Sentry integration with Expo allows you to view crash reports and Session Replays for your Expo app deployments directly within your Expo dashboard. This integration provides a direct link to Sentry stack traces with full context, session replays, and debugging capabilities.
```

### Install

> Sentry owner, manager, or admin permissions are required to install this integration.

> 1. Log into EAS and open **Account Settings > Overview** in EAS (e.g., `https://expo.dev/accounts/[your-account]/settings`).

Reviewer: ```suggestion
1. Log in into you Expo account and open [**Account settings > Overview**](https://expo.dev/accounts/[your-account]/settings).
```

Let's update the verbiage for consistency and link the settings path so that already logged-in users can open the link directly

> 2. Locate the **Connections** section and click **Connect** next to Sentry.

Reviewer: ```suggestion
2. Under **Connections** and click **Connect** next to Sentry.
```

> 3. Login to Sentry and accept the integration into your organization. You will be redirected back to Account Settings > Overview.

Reviewer: ```suggestion
3. Log in to your Sentry account and accept the integration into your organization. You will be redirected back to **Account settings**.
```

> ### Configure
> 
> #### Link Your Projects

Reviewer: ```suggestion
### Link your project
```

This can be a single section.

After connecting your accounts, you need to link your EAS Project to your Sentry Project:

1. Open **Projects > [Your Project] > Configuration > Project settings** in EAS.
2. Click **Link** and select your Sentry Project from the dropdown.

### Usage

> To see your Sentry data in EAS:

Reviewer: ```suggestion
To see your Sentry data in Expo dashboard:
```

1. Open **Projects > [Your Project] > Updates > Deployments > [Deployment]** to view Sentry data from a Release.

With this integration, you can:

> - View crash reports directly in your EAS dashboard

Reviewer: ```suggestion
- View crash reports directly in your Expo dashboard
```
